.github/workflows/ci.yml

```diff
@@ -674,6 +686,129 @@ jobs:
               body: 'Failed to pass the component examples UI Test. If this PR is for UI change and the error is snapshot mismatch, please add "update_snapshots" label to the PR for updating the snapshot.'
             })
 
+  compare_jest_tests_coverage:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request' }}
+    name: Compare unit tests coverage from pr head ref to base ref (${{ matrix.flavor }})
+    needs: build_packages
+    strategy:
+      matrix:
+        include:
+          - flavor: stable
+            gist_id: d1528e50153e6f7ea569040d7b437abb
+          - flavor: beta
+            gist_id: 67224b1799c92f0e1cd2da24798e4dd5
+    steps:
+      - name: Get jest tests coverage of main from gist
+        run: curl -o temp/jest/base/coverage-report.json "https://gist.github.com/communication-ui-bot/${{ matrix.gist_id }}/raw/communication-react-jest-report-${{ matrix.flavor }}.json" --create-dirs -L
+      - name: Download coverage report from the build_packages job
+        uses: actions/download-artifact@v3
+        with:
+          name: jest-coverage-report-${{ matrix.flavor }}
+          path: temp/jest/current/merge/
+      - name: Calculate coverage change
+        id: coverage
+        run: |
+          base_lines_covered=$(jq -r ".total.lines.covered" temp/jest/base/coverage-report.json)
```

Comment:
this is a lot in a yaml file (which are notoriously hard to test/debug/reuse) -- can we move this to a script under `common/scripts`?

Reply:
Updated, but the script is added under workflow folder because of dependencies install